sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp

```diff
@@ -0,0 +1,2630 @@
+        {
+          response.ClientRequestId = response_clientrequestid_iterator->second;
+        }
+        response.ETag = http_response.GetHeaders().at("ETag");
+        response.LastModified = http_response.GetHeaders().at("Last-Modified");
+        for (auto i = http_response.GetHeaders().lower_bound("x-ms-meta-");
+             i != http_response.GetHeaders().end() && i->first.substr(0, 10) == "x-ms-meta-";
+             ++i)
+        {
+          response.Metadata.emplace(i->first.substr(10), i->second);
+        }
+        auto response_accesstype_iterator
+            = http_response.GetHeaders().find("x-ms-blob-public-access");
+        if (response_accesstype_iterator != http_response.GetHeaders().end())
+        {
+          response.AccessType = PublicAccessTypeFromString(response_accesstype_iterator->second);
+        }
+        response.HasImmutabilityPolicy
+            = http_response.GetHeaders().at("x-ms-has-immutability-policy") == "true";
+        response.HasLegalHold = http_response.GetHeaders().at("x-ms-has-legal-hold") == "true";
+        response.LeaseStatus
+            = BlobLeaseStatusFromString(http_response.GetHeaders().at("x-ms-lease-status"));
+        response.LeaseState
+            = BlobLeaseStateFromString(http_response.GetHeaders().at("x-ms-lease-state"));
+        auto response_leaseduration_iterator
+            = http_response.GetHeaders().find("x-ms-lease-duration");
+        if (response_leaseduration_iterator != http_response.GetHeaders().end())
+        {
+          response.LeaseDuration = response_leaseduration_iterator->second;
+        }
+        return response;
+      }
+
+      static BlobContainerProperties GetProperties(
+          Azure::Core::Context context,
+          Azure::Core::Http::HttpPipeline& pipeline,
+          const std::string& url,
+          const GetPropertiesOptions& options)
+      {
+        auto request = GetPropertiesConstructRequest(url, options);
+        auto response = pipeline.Send(context, request);
+        return GetPropertiesParseResponse(*response);
+      }
+
+      struct SetMetadataOptions
+      {
+        std::string Version;
+        std::string Date;
+        std::map<std::string, std::string> Metadata;
+      }; // struct SetMetadataOptions
+
+      static Azure::Core::Http::Request SetMetadataConstructRequest(
+          const std::string& url,
+          const SetMetadataOptions& options)
+      {
+        auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Put, url);
+        request.AddHeader("Content-Length", "0");
+        request.AddQueryParameter("restype", "container");
+        request.AddQueryParameter("comp", "metadata");
+        request.AddHeader("x-ms-version", options.Version);
+        request.AddHeader("x-ms-date", options.Date);
+        for (const auto& pair : options.Metadata)
+        {
+          request.AddHeader("x-ms-meta-" + pair.first, pair.second);
```

Comment:
>request.AddHeader("x-ms-meta-" + pair.first, pair.second); [](start = 10, length = 58)

What's the behavior when customer specifies `"key": "value"` and `"KEY": "value2"`?

Reply:
exception